backend-api/src/functions/asyncActiveSession/tokenService/publicKeyGetter.ts

```diff
@@ -5,36 +5,46 @@ import {
   ProtectedHeaderParameters,
 } from "jose";
 import { errorResult, Result, successResult } from "../../utils/result";
+import {
+  ISendHttpRequest,
+  sendHttpRequest,
+} from "../../services/http/sendHttpRequest";
+
+export type PublicKeyGetterDependencies = {
+  sendHttpRequest: ISendHttpRequest;
+};
+
+const publicKeyGetterDependencies: PublicKeyGetterDependencies = {
+  sendHttpRequest: sendHttpRequest,
+};
 
 export class PublicKeyGetter implements IPublicKeyGetter {
+  private readonly sendHttpRequest: ISendHttpRequest;
+  constructor(dependencies = publicKeyGetterDependencies) {
+    this.sendHttpRequest = dependencies.sendHttpRequest;
+  }
+
   async getPublicKey(
-    jwks: IJwks,
+    jwksEndpoint: string,
     jwt: string,
   ): Promise<Result<Uint8Array | KeyLike>> {
-    let decodedProtectedHeader: ProtectedHeaderParameters;
+    let kid;
     try {
-      decodedProtectedHeader = decodeProtectedHeader(jwt);
+      kid = this.getTokenKid(jwt);
     } catch (error) {
       return errorResult({
-        errorMessage: `${error}`,
-        errorCategory: "CLIENT_ERROR",
-      });
-    }
-
-    const keyId = decodedProtectedHeader.kid;
-    if (!keyId) {
-      return errorResult({
-        errorMessage: "kid not present in JWT header",
+        errorMessage: `Error getting public key: ${error}`,
         errorCategory: "CLIENT_ERROR",
       });
     }
 
-    const jwk = jwks.keys.find((key) => key.kid && key.kid === keyId);
-
-    if (!jwk) {
+    let jwk;
+    try {
+      jwk = await this.getJwk(jwksEndpoint, kid);
+    } catch (error) {
       return errorResult({
-        errorMessage: "JWKS does not contain key matching provided key ID",
-        errorCategory: "CLIENT_ERROR",
+        errorMessage: `Error getting public key: ${error}`,
```

Comment:
SERVER_ERROR when we can't fetch keys from STS or the keys we get back are malformed or do not contain a key that matches our JWT's kid